src/v2/Components/Onboarding/Steps/Artists/PopularArtists.tsx

```diff
@@ -59,11 +59,13 @@ class PopularArtistsContent extends React.Component<Props, null> {
     )
     this.excludedArtistIds.add(artistToSuggest.getValue("internalID") as string)
 
-    const popularArtistsRootField = store
-      .get("client:root")
-      .getLinkedRecords("popular_artists", { exclude_followed_artists: true })
+    const popularArtistsRootField = store.get("client:root")
+    const popularArtists =
+      popularArtistsRootField.getLinkedRecords("popular_artists", {
+        exclude_followed_artists: true,
+      }) || []
 
-    const updatedPopularArtists = popularArtistsRootField
+    const updatedPopularArtists = popularArtists
```

Comment:
wondering whats up with all of this store manipulation stuff, why it was originally necessary, we don't do this elsewhere really

Reply:
Yeah! I was super confused by it and there was a bug there but I couldn't really make out what was happening. 🤷